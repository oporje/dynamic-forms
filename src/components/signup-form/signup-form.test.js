import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import React from 'react';
import { shallow, mount } from 'enzyme';
import SignUp from './signup-form';

Enzyme.configure({adapter: new Adapter()});

test('renders the sign up component', () => {
  const wrapper = shallow(<SignUp />);
  expect(wrapper.find('#submitBtn').exists()).toBeTruthy();
  expect(wrapper.find('#resetBtn').exists()).toBeTruthy();
  expect(wrapper).toMatchSnapshot();
});

test('check form validations', () => {
    const wrapper = mount(<SignUp />);
    expect(wrapper.find('.text-input_fieldWrapper .errorMessage').exists()).toBeFalsy();
    const nameInput = wrapper.find('input#firstName');
    nameInput.simulate("change", { target: { value: "foo" }});
    const submitBtn = wrapper.find("#submitBtn");
    submitBtn.simulate('click');
});
  

