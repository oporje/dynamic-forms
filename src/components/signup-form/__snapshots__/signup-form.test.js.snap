// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `ShallowWrapper {}`;

exports[`renders the sign up component 1`] = `ShallowWrapper {}`;

exports[`renders the sign up wrapper 1`] = `ShallowWrapper {}`;
